--- a/emsdk/upstream/4583/sysroot/lib/libc.imports	2019-04-01 09:45:26.467798904 +0100
+++ b/emsdk/upstream/4583/sysroot/lib/libc.imports	2019-04-01 09:46:08.527195096 +0100

@@ -29,3 +29,40 @@
 # Part of musl we still don't compile for wasm
 setjmp
 longjmp
+
+# Dynamic linking
+dlopen
+dlsym
+dlclose
+dlerror
+
+# Misc
+ioctl
+
+# Python-related
+realpath
+zlibVersion
+deflateEnd
+deflate
+inflateEnd
+inflate
+inflateSetDictionary
+adler32
+deflateInit_
+deflateInit2_
+deflateSetDictionary
+crc32
+inflateInit2_
+getpriority
+setpriority
+uname
+openpty
+forkpty
+lockf
+getresuid
+getresgid
+getrusage
+getrlimit
+setrlimit
+__copy_tls
+
