--- a/emsdk/upstream/4138/sysroot/lib/libc.imports	2019-03-28 14:37:04.828279429 +0000
+++ b/emsdk/upstream/4138/sysroot/lib/libc.imports	2019-03-28 14:37:04.828279429 +0000
@@ -30,3 +30,5 @@
 setjmp
 longjmp
 
+# Dynamic linking
+dlopen
+dlsym
+dlclose
+dlerror
+
+# Misc
+ioctl
+
+# Python-related
+realpath
+zlibVersion
+deflateEnd
+deflate
+inflateEnd
+inflate
+inflateSetDictionary
+adler32
+deflateInit_
+deflateInit2_
+deflateSetDictionary
+crc32
+inflateInit2_
+getpriority
+setpriority
+uname
+openpty
+forkpty
+lockf
+getresuid
+getresgid
+getrusage
+getrlimit
+setrlimit
+__copy_tls
+

