diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index c3682b4..7391ba9 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -3548,9 +3548,7 @@ _posix_listdir(path_t *path, PyObject *list)
 
         return_str = 1;
 
-        Py_BEGIN_ALLOW_THREADS
         dirp = fdopendir(fd);
-        Py_END_ALLOW_THREADS
     }
     else
 #endif
@@ -3566,18 +3564,14 @@ _posix_listdir(path_t *path, PyObject *list)
             return_str = 1;
         }
 
-        Py_BEGIN_ALLOW_THREADS
         dirp = opendir(name);
-        Py_END_ALLOW_THREADS
     }
 
     if (dirp == NULL) {
         list = path_error(path);
 #ifdef HAVE_FDOPENDIR
         if (fd != -1) {
-            Py_BEGIN_ALLOW_THREADS
             close(fd);
-            Py_END_ALLOW_THREADS
         }
 #endif
         goto exit;
@@ -3587,9 +3581,7 @@ _posix_listdir(path_t *path, PyObject *list)
     }
     for (;;) {
         errno = 0;
-        Py_BEGIN_ALLOW_THREADS
         ep = readdir(dirp);
-        Py_END_ALLOW_THREADS
         if (ep == NULL) {
             if (errno == 0) {
                 break;
@@ -3621,13 +3613,11 @@ _posix_listdir(path_t *path, PyObject *list)
 
 exit:
     if (dirp != NULL) {
-        Py_BEGIN_ALLOW_THREADS
 #ifdef HAVE_FDOPENDIR
         if (fd > -1)
             rewinddir(dirp);
 #endif
         closedir(dirp);
-        Py_END_ALLOW_THREADS
     }
 
     return list;
